core/src/test/scala/io/projectglow/vcf/VariantContextToInternalRowConverterSuite.scala

```diff
@@ -175,6 +175,33 @@ class VariantContextToInternalRowConverterSuite extends VCFConverterBaseTest {
     assert(vcfRow == convertedVcWithDefaultGt)
   }
 
+  test("Checks runtime datatype for format array fields") {
+    val vcb = new VariantContextBuilder()
+    vcb.chr("").start(1).stop(1)
+    val refAllele = Allele.create("A", true)
+    vcb.alleles(Seq(refAllele).asJava)
+
+    def expectFailure(gp: Any): Unit = {
+      val vc = vcb.genotypes(new GenotypeBuilder().attribute("GP", gp).make()).make()
+      intercept[IllegalArgumentException] {
+        strictConverter.convertRow(vc, isSplit = false)
+      }
+    }
+
+    expectFailure(new JArrayList[Integer](Seq(1: Integer).asJavaCollection))
+    expectFailure(Array[Integer](1: Integer))
+    expectFailure(Array[Int](1))
```

Comment:
Let's also add a success case as a sanity check, and to clarify why these are failing.

Reply:
That's why I added the following test